### PR TITLE
IGNITE-20105 Set default precision and scale for new column descriptor when needed.

### PR DESCRIPTION
**Pull request by AMashenkov**
https://issues.apache.org/jira/browse/IGNITE-20105

### CODE REVIEW

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/ColumnParams.java [hunk 1]
```diff
@@ -158,7 +158,7 @@ public Builder defaultValue(DefaultValue defaultValue) {
          *
          * @return {@code this}.
          */
-        public Builder precision(int precision) {
+        public Builder precision(Integer precision) {
```
Comment (tkalkirill) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/ColumnParams.java:161]: Please add `@Nullable` for method `org.apache.ignite.internal.catalog.commands.ColumnParams#precision` and specify in documentation what `null` means.

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/ColumnParams.java [hunk 2]
```diff
@@ -169,7 +169,7 @@ public Builder precision(int precision) {
          *
          * @return {@code this}.
          */
-        public Builder scale(int scale) {
+        public Builder scale(Integer scale) {
```
Comment (tkalkirill) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/ColumnParams.java:172]: Please add `@Nullable` for method `org.apache.ignite.internal.catalog.commands.ColumnParams#scale` and specify in documentation what `null` means.

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/ColumnParams.java [hunk 3]
```diff
@@ -180,7 +180,7 @@ public Builder scale(int scale) {
          *
          * @return {@code this}.
          */
-        public Builder length(int length) {
+        public Builder length(Integer length) {
```
Comment (tkalkirill) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/ColumnParams.java:183]: Please add `@Nullable` for method `org.apache.ignite.internal.catalog.commands.ColumnParams#length` and specify in documentation what `null` means.

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java [hunk 4]
```diff
@@ -155,9 +204,10 @@ public static CatalogZoneDescriptor fromParams(int id, CreateZoneParams params)
      * @return Column descriptor.
      */
     public static CatalogTableColumnDescriptor fromParams(ColumnParams params) {
-        int precision = params.precision() != null ? params.precision() : 0;
-        int scale = params.scale() != null ? params.scale() : 0;
-        int length = params.length() != null ? params.length() : 0;
+        int precision = params.precision() != null ? params.precision() : defaultPrecision(params.type());
```
Comment (tkalkirill) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java:207]: Maybe use **java.util.Objects#requireNonNullElse** ? =)

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java [hunk 5]
```diff
@@ -176,4 +226,36 @@ public static boolean isSupportedColumnTypeChange(ColumnType source, ColumnType
...
+            case BITMASK:
+            case STRING:
+            case BYTE_ARRAY:
+                return Integer.MAX_VALUE;
+            default:
+                /*
+                 * Default length is 1 if implicit.
+                 * SQL`16 part 2 section 6.1 syntax rule 5
+                 */
+                return Math.max(1, defaultPrecision(columnType));
```
Comment (tkalkirill) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java:258]: I think it's worth putting `1` into a constant with a description.

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java (hunk 4 above)
Comment (AMashenkov) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java:207]: 1. java.util.Objects#requireNonNullElse will force argument evaluation.

Comment (tkalkirill) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java:214]: At your discretion, this operation does not seem to be performed often, and you can omit this nuance.

**Pull request merged**